static/docs/user-guide/using-docker.md

```diff
@@ -0,0 +1,108 @@
+# Using Docker
+
+Currently we don't provide any official Docker image, however you can easily
+create it yourself.
+
+## Writing the Dockerfile
+
+We support several ways to install DVC, the two recommended ones are through
+`pip` and `conda`.
+
+You can base your image on the official Python one:
+```dockerfile
+FROM python
+
+# Remember to include the cloud providers that you plan to use,
+# for example: "dvc[s3]"
+RUN pip install dvc
+```
+
+Or use a conda based image:
+```dockerfile
+FROM continuumio/miniconda3
+
+RUN conda install dvc -y --quiet
+```
+
+For a recommended reading about writing Dockerfiles for Python applications,
+you can take a look at https://pythonspeed.com/docker/
+
+## Using Docker for your development environment
+
+Since `dvc` is a command line application, we have several recommendiations
+to enhance the experience:
+  * Make sure your _locale_ is set to UTF-8
+  * Install the `Zsh` shell and its respective autocompletion script.
+  * Install `less` so you can pipe the output of `dvc pipeline show`.
+  * Enable movements with `Ctrl + Arrow` by editing the _inputrc_ file.
+  * Use `pip install dvc` at the end of the file for better caching.
+  * Set your `WORKDIR` to the path where you are going to mount your project.
+
+If you are using Linux, remember to map your user to your container. This way,
+you make sure your files are still writtable. For more information:
+https://docs.docker.com/engine/security/userns-remap/
+
+### Dockerfile
```

Comment:
would be beneficial to just provide a link to the file and put it into Git so that we can iterate with a regular process?

the section should be more focused on philosophy (like above ^^) that explains certain decision, how to start using it, proving links, etc

Reply:
> would be beneficial to just provide a link to the file and put it into Git so that we can iterate with a regular process?

for the user I don't think it would bring significant value, and for us, well, I don't mind keeping the Dockerfile example on the docs)

> the section should be more focused on philosophy (like above ^^) that explains certain decision, how to start using it, proving links, etc

Reply:
> for the user I don't think it would bring significant value

At least it's just easier to get that file, instead of copying it. It's the way code files are usually distributed, right? :)

Reply:
@shcheklein , all right, let's do this)

where should I put the example Dockerfile

Reply:
DVC core? At least it's a better place than the website and docs :) Let's imagine we decide to package an image via CI - we will be doing it probably in DVC core, or in a separate repo. Probably not in dvc.org.

Reply:
Still not sure that we want to do this, but let's try, anyways :sweat_smile: